project-base/tests/ReadModelBundle/Functional/Twig/ImageExtensionTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\ReadModelBundle\Functional\Twig;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Image\AdditionalImageData;
+use Shopsys\FrameworkBundle\Component\Image\ImageFacade;
+use Shopsys\FrameworkBundle\Component\Image\ImageLocator;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Twig\ImageExtension;
+use Tests\ShopBundle\Test\FunctionalTestCase;
+
+class ImageExtensionTest extends FunctionalTestCase
+{
+    public function testGetImageHtmlWithMockedImageFacade(): void
+    {
+        $domain = $this->getContainer()->get(Domain::class);
+        $imageLocator = $this->getContainer()->get(ImageLocator::class);
+        $templating = $this->getContainer()->get('templating');
+
+        $productId = 2;
+        $entityName = 'product';
+        $fileExtension = 'jpg';
+
+        $imageFacadeMock = $this->createMock(ImageFacade::class);
+        $imageFacadeMock->method('getImageUrlFromAttributes')->willReturn(sprintf('http://webserver:8080/%s/%d.%s', $entityName, $productId, $fileExtension));
+        $imageFacadeMock->method('getAdditionalImagesDataFromAttributes')->willReturn([
+            new AdditionalImageData('(min-width: 1200px)', sprintf('http://webserver:8080/%s/additional_0_%d.%s', $entityName, $productId, $fileExtension)),
+            new AdditionalImageData('(max-width: 480px)', sprintf('http://webserver:8080/%s/additional_1_%d.%s', $entityName, $productId, $fileExtension)),
+        ]);
+
+        $imageView = new ImageView($productId, $fileExtension, $entityName, null);
+
+        $readModelBundleImageExtension = new ImageExtension('', $domain, $imageLocator, $imageFacadeMock, $templating);
+
+        $html = $readModelBundleImageExtension->getImageHtml($imageView);
+
+        $this->assertXmlStringEqualsXmlFile(__DIR__ . '/Resources/picture.twig', $html);
+
+        libxml_clear_errors();
+    }
+
+    public function testGetImageHtml(): void
+    {
+        $domain = $this->getContainer()->get(Domain::class);
+        $imageLocator = $this->getContainer()->get(ImageLocator::class);
+        $templating = $this->getContainer()->get('templating');
+
+        $imageFacade = $this->getContainer()->get(ImageFacade::class);
+
+        $productId = 1;
+        $entityName = 'product';
+        $fileExtension = 'jpg';
+
+        $imageView = new ImageView($productId, $fileExtension, $entityName, null);
+
+        $readModelBundleImageExtension = new ImageExtension('', $domain, $imageLocator, $imageFacade, $templating);
```

Comment:
At this point you can even get the `ImageExtension` from container, right? (maybe not because of the prefix, but I hope so). Same applies for following tests.

If you cannot receive the extension from container, please create method that will create the the object (and remove duplicits)

Reply:
I created a method and simplified test cases with it.